language/move-core/types/src/language_storage.rs

```diff
@@ -18,34 +18,58 @@ use std::{
 pub const CODE_TAG: u8 = 0;
 pub const RESOURCE_TAG: u8 = 1;
 
-const MAX_TYPE_TAG_NESTING: u8 = 8;
+pub const MAX_TYPE_TAG_NESTING: u8 = 8;
+
+#[cfg(test)]
+const MAX_DE_TYPE_TAG_NESTING: u8 = MAX_TYPE_TAG_NESTING - 1;
```

Comment:
to confirm, this is there just so we can check that both serialization and deserialization fails, correct?

wanna add a comment for that?